### PR TITLE
added redis cache container

### PR DESCRIPTION
**Pull request by Devesh21700Kumar**
## Pull request checklist

Please check if your PR :rocket: fulfills the following requirements:
- [x] Build (`npm run start`) was run locally and new changes were working correctly
- [x] Lint (`npm run linter`) has passed locally and any fixes were made for failures

Please check the type of change your PR introduces:
- [x] Bugfix
- [x] Feature
- [x] Refactoring (no functional changes, no api changes)
- [x] Build related changes

## What is the current behavior?

redis cache container setup and works with both npm start and docker-compose up

Fixes #

#19

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -1,38 +1,44 @@
...
+      - /usr/src/app/node_modules
+    networks:
+      - webnet
+    depends_on:
+      - redis
+      - postgres
+    working_dir: /usr/src/app
+  redis:
+    container_name: redis
+    image: redis:5
```
Comment (Sarvesh67) [inline docker-compose.yml:22]: Use Redis 6 image

File: docker-compose.yml [hunk 2]
```diff
@@ -1,38 +1,44 @@
...
+    volumes:
+      - ./:/usr/src/app/
+      - /usr/src/app/node_modules
+    networks:
+      - webnet
+    depends_on:
+      - redis
+      - postgres
+    working_dir: /usr/src/app
+  redis:
```
Comment (Sarvesh67) [inline docker-compose.yml:20]: No volumes for persisting data

File: src/shared/services/config.service.ts [hunk 3]
```diff
@@ -33,7 +33,7 @@ export class ConfigService {
 
     get typeOrmConfig(): TypeOrmModuleOptions {
         return {
-            host: this.isDevelopment ? 'localhost' : this.get('DB_HOST') || 'postgres',
+            host: this.isDevelopment ? 'localhost' || 'postgres' : this.get('DB_HOST') || 'postgres',
```
Comment (Sarvesh67) [inline src/shared/services/config.service.ts:36]: What does this change do? Either implement config taking docker env into consideration or revert the change.

File: src/shared/services/config.service.ts [hunk 4]
```diff
@@ -51,7 +51,7 @@ export class ConfigService {
     get redis() {
         return {
             port: this.getNumber('REDIS_PORT') || 6439,
-            host: this.get('REDIS_HOST') || 'localhost'
+            host: this.get('REDIS_HOST') || 'localhost' || 'postgres'
```
Comment (Sarvesh67) [inline src/shared/services/config.service.ts:54]: What does this change do? Either implement config taking docker env into consideration or revert the change.

**Review by Sarvesh67 — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (Devesh21700Kumar) [reply docker-compose.yml:22]: k

File: src/shared/services/config.service.ts (hunk 3 above)
Comment (Devesh21700Kumar) [reply src/shared/services/config.service.ts:36]: set connection for typeorm with postgres while using docker-compose in development as well

File: src/shared/services/config.service.ts (hunk 4 above)
Comment (Devesh21700Kumar) [reply src/shared/services/config.service.ts:54]: ok will delete this
